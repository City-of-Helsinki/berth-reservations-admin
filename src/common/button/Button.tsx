import React from 'react';
import classNames from 'classnames';

import styles from './button.module.scss';

export type Props = {
  color?: 'standard' | 'brand' | 'critical' | 'secondary' | 'info';
  variant?: 'contained' | 'outlined' | 'text';
  size?: 'small' | 'standard' | 'large';
  icon?: JSX.Element;
} & React.DOMAttributes<HTMLButtonElement>;

const Button: React.SFC<Props> = ({
  children,
  color = 'standard',
  variant = 'contained',
  size = 'standard',
  icon,
  onClick,
}) => {
  return (
    <button
      type="button"
      onClick={onClick}
      className={classNames(
        styles.button,
        styles[color],
        size && styles[size],
        variant && styles[variant],
        icon && children && styles.hasIcon
      )}
    >
      {icon}
      {children}
    </button>
  );
};

export default Button;
