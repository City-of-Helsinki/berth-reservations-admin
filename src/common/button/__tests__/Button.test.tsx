import React from 'react';
import { shallow } from 'enzyme';

import Button from '../Button';

describe('Button', () => {
  const getWrapper = () => shallow(<Button>Test</Button>);

  it('renders normally', () => {
    const wrapper = getWrapper();

    expect(wrapper.render()).toMatchSnapshot();
  });
});
