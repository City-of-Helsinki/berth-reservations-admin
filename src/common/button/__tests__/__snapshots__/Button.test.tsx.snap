// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders normally 1`] = `
<button
  class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO Button-module_theme-coat__2_3U7 button_hds-button--theme-coat__12cO0"
  type="button"
>
  <span
    class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
  >
    Test
  </span>
</button>
`;
